lib/middlewares/auth.js

```diff
@@ -77,3 +79,20 @@ passport.use(
     }
   })
 );
+
+export const permitirAcceso = (...roles) => {
+  return async (req, res, next) => {
+    const token = req.headers['authorization'];
+    const decodeToken = jwt_decode(token);
```

Comment:
Me llama la atención la inclusión de este nueva biblioteca. ¿Si antes no estaba cómo se leía el token?

Reply:
El tema es que queríamos sacar la info que traía el token porque sino no sabíamos cómo pasarle el usuario... Y si en el token se guarda el id del usuario, lo podíamos sacar de ahí. Entonces decodificamos el token y buscamos en la base el usuario para saber que rol tiene y así bocharlo o darle acceso

Reply:
Se podía hacer a mano con una [función mágica](https://stackoverflow.com/questions/38552003/how-to-decode-jwt-token-in-javascript-without-using-a-library) o usar esa librería y nos pareció más prolija esta última opción.

Reply:
Ojo, estoy de acuerdo con usar una biblioteca, lo que digo es que me llama la atención que hayan tenido que incluir otra cuando ya teníamos `jsonwebtoken`. Ahí miré un poco la doc y encontré un método [`jwt.decode`](https://www.npmjs.com/package/jsonwebtoken#jwtdecodetoken--options), ¿podrás probar si ese funca?

En ese caso, podríamos prescindir de `jwt_decode`.

Reply:
Si, en un principio utilizamos la de `jsonwebtoken`, pero el `req.headers['authorization'];` viene con el `Bearer ` al comienzo y esta biblioteca no lo decodificaba. Teníamos que extraer solo el token con un `split(' ')[1]`, por eso decidimos agregar `jwt_decode`, donde esto no hacía falta.

Reply:
Ok, mergeamos entonces.